includes/Endpoints/CreatePost.php

```diff
@@ -2,24 +2,130 @@
 
 namespace VPlugins\SMPostConnector\Endpoints;
 
+use WP_REST_Request;
+use WP_REST_Response;
+use VPlugins\SMPostConnector\Middleware\AuthMiddleware;
+use VPlugins\SMPostConnector\Helper\Globals;
+
 class CreatePost {
     public function __construct() {
+        $this->auth_middleware = new AuthMiddleware();
```

Comment:
what auth are we using here?

Reply:
In the SM Post Connector plugin, we are using a token-based authentication method rather than the regular SSO/WP login. We create an access token and save it in the database. Each time a request is made to our API endpoints, we check this token to validate access. This ensures that only authorized requests can interact with the SM API.

The `AuthMiddleware` class handles this authentication process by verifying the token included in the request. The request can proceed if the token is valid and matches the one stored in the database; otherwise, it is denied.